src/beast/beast/nudb/detail/bucket.h

```diff
@@ -297,54 +250,80 @@ bucket_t<_>::at (std::size_t i) const ->
         is, result.offset);             // Offset
     detail::read<uint48_t>(
         is, result.size);               // Size
-    result.key = is.data(key_size_);    // Key
+    detail::read<hash_t>(
+        is, result.hash);               // Hash
     return result;
 }
 
 template <class _>
-auto
-bucket_t<_>::find (void const* key) const ->
-    std::pair<value_type, bool>
+std::size_t
+bucket_t<_>::lower_bound (
+    std::size_t h) const
 {
-    std::pair<value_type, bool> result;
-    std::size_t i;
-    std::tie(i, result.second) = lower_bound(key);
-    if (result.second)
-        result.first = at(i);
-    return result;
+    // Bucket Entry
+    auto const w =
+        field<uint48_t>::size +         // Offset
+        field<uint48_t>::size +         // Size
+        field<hash_t>::size;            // Hash
+    // Bucket Record
+    auto const p = p_ +
+        field<std::uint16_t>::size +    // Count
+        field<uint48_t>::size +         // Spill
+        // Bucket Entry
+        field<uint48_t>::size +         // Offset
+        field<uint48_t>::size;          // Size
+    std::size_t step;
+    std::size_t first = 0;
+    std::size_t count = size_;
+    while (count > 0)
+    {
+        step = count / 2;
+        auto const i = first + step;
+        std::size_t h1;
+        readp<hash_t>(p + i * w, h1);
+        if (h1 < h)
+        {
+            first = i + 1;
+            count -= step + 1;
+        }
+        else
+        {
+            count = step;
+        }
+    }
+    return first;
 }
```

Comment:
What about using an iterator based on `p` and `w' and a comparator and using `std::lower_bound`? It's easier to confirm that just the iterator and comparator are correct.

FWIW, the code above looks correct.